### PR TITLE
manifest loading performance improvements

### PR DESCRIPTION
**Pull request by tomerd**
motivation: better performance for dependency resolution

changes:
* make the loadManifest function asynchronous
* move the operation queue into ManifestLoader instead of Workspace
* load root manifest asynchronous
* adjust call sites
* adjust test
* replace some queue.wait with queue.notify
* tested with TSAN

### CODE REVIEW

**Comment by tomerd** [comment]
note the second commit is only formatting, no functional changes

**Comment by tomerd** [comment]
@swift-ci please smoke test

File: Sources/PackageLoading/ManifestLoader.swift [hunk 1]
```diff
@@ -233,39 +274,70 @@ public final class ManifestLoader: ManifestLoaderProtocol {
...
+                          version: version,
+                          revision: revision,
+                          toolsVersion: toolsVersion,
+                          packageIdentity: packageIdentity,
+                          packageKind: packageKind,
+                          fileSystem: fileSystem,
+                          diagnostics: diagnostics,
+                          on: queue) { result in
+
+                // cache positive results
```
Comment (abertelrud) [inline Sources/PackageLoading/ManifestLoader.swift:331]: This is great that we're no longer caching failures.  That will help debugging especially (in real use I think most fixes involve changes to the manifest so the caching of failures might be less of a problem there today).

**Review by abertelrud — APPROVED** [review]
This is great!

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Pull request merged**